### ISSUE TITLE
Not so rare anymore game crash

### ISSUE THREAD

**Issue by Dalton-Baird** [EOUCS344Sp15/ReleaseTheKraken#53]
The game has been observed to freeze and crash a few times, but it seems to be random and rare.  So far, I am unaware of the cause of this, but it started happening around the time we added projectile firing support.  The log just stops when it happens, which isn't much of a help.  

If you encounter this and/or have information about it, please comment.

**Comment by Dalton-Baird** [comment]
I just had a crash happen, and something actually got printed to the log.  Looks like it has to do with Box2D...

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x0000000066bcbd0d, pid=4304, tid=2416
#
# JRE version: Java(TM) SE Runtime Environment (8.0_25-b18) (build 1.8.0_25-b18)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.25-b02 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# C  [gdx-box2d64.dll+0xbd0d]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# D:\Documents\NetBeansProjects\ReleaseTheKraken\core\assets\hs_err_pid4304.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
```

**Comment by Dalton-Baird** [comment]
I've been trying to find out what is causing this.  There are two separate crashes that happen:
1. The game freezes, and you have to close it.  No log output.
2. The game crashes, printing the EXCEPTION_ACCESS_VIOLATION message to the log (Windows).  It seems to always mention Box2D.  In the error logs that get generated, it mentions that it was removing a physics body.  

I looked it up, and found out that Box2D will crash if you try to remove a physics body that has already been removed.  That might be what is causing the problem.  It could be that two projectiles are colliding with an enemy at once, both delivering the final blow and adding it to the list of entities to remove twice.  I'll look into it more to see if I can fix it.

**Comment by Dalton-Baird** [comment]
That was actually the problem!  So far, I've had it running for a while, and it hasn't crashed yet.  I made it only add the entities to delete to the list if they didn't already exist in it.  If they did, I made it print an error message.  There were tons of error messages in the log, so that was what was happening.  I haven't pushed this change yet, but I will soon.

Another thing... I think the pirate base might be a bit too tough...
![screenshot-51](https://cloud.githubusercontent.com/assets/12261912/7906144/ed0c0ade-07d9-11e5-9b0a-3ec7bef4faae.png)
